engine/src/main/java/org/terasology/persistence/typeHandling/mathTypes/BlockAreaTypeHandler.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.persistence.typeHandling.mathTypes;
+
+import com.google.common.collect.Maps;
+import gnu.trove.list.TIntList;
+import org.terasology.persistence.typeHandling.PersistedData;
+import org.terasology.persistence.typeHandling.PersistedDataArray;
+import org.terasology.persistence.typeHandling.PersistedDataMap;
+import org.terasology.persistence.typeHandling.PersistedDataSerializer;
+import org.terasology.persistence.typeHandling.TypeHandler;
+import org.terasology.world.block.BlockArea;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class BlockAreaTypeHandler extends TypeHandler<BlockArea> {
+    private static final String MIN_FIELD = "min";
+    private static final String MAX_FIELD = "max";
+    private static final String SIZE_FIELD = "size";
+
+    @Override
+    protected PersistedData serializeNonNull(BlockArea value, PersistedDataSerializer serializer) {
+        Map<String, PersistedData> map = Maps.newLinkedHashMap();
+        map.put(MIN_FIELD, serializer.serialize(value.minX(), value.minY()));
+        map.put(MAX_FIELD, serializer.serialize(value.maxX(), value.maxY()));
+        return serializer.serialize(map);
+    }
+
+    @Override
+    public Optional<BlockArea> deserialize(PersistedData data) {
+        if (!data.isNull() && data.isValueMap()) {
+            PersistedDataMap map = data.getAsValueMap();
+
+            PersistedDataArray minDataArr = map.get(MIN_FIELD).getAsArray();
+            TIntList minArr = minDataArr.getAsIntegerArray();
+            if (map.has(SIZE_FIELD)) {
+                PersistedDataArray sizedataArray = map.get(SIZE_FIELD).getAsArray();
+                TIntList sizeArr = sizedataArray.getAsIntegerArray();
+                return Optional.of(
+                    new BlockArea(
+                        minArr.get(0), minArr.get(1)).setSize(
+                        minArr.get(0) + sizeArr.get(0),
+                        minArr.get(1) + sizeArr.get(1)));
```

Comment:
As you are using 'setSize(...)` you don't need to do any computation here:
```suggestion
                    new BlockArea(minArr.get(0), minArr.get(1))
                    		.setSize(sizeArr.get(0), sizeArr.get(1)));
```